### PR TITLE
Fix validation on access and sale modal

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description

Fixes release-blocking issue with access and sale modal.

Fixes
- Form was not able to be submitted when a validation error was present in other hidden fields
- Cleans form values when swapping between availability types

### How Has This Been Tested?
local web

### CODE REVIEW

File: packages/web/src/pages/upload-page/fields/AccessAndSaleField.tsx [hunk 1]
```diff
@@ -140,26 +140,47 @@ export type AccessAndSaleFormValues = {
...
+      },
+      { message: messages.errors.preview.tooEarly, path: [PREVIEW] }
+    )
+    .refine(
+      (values) => {
+        const formValues = values as AccessAndSaleFormValues
+        if (isPremiumContentUSDCPurchaseGated(formValues[PREMIUM_CONDITIONS])) {
+          return (
+            formValues[PREVIEW] === undefined ||
+            (formValues[PREVIEW] >= 0 && formValues[PREVIEW] < trackLength - 15)
```
Comment (rickyrombo) [inline packages/web/src/pages/upload-page/fields/AccessAndSaleField.tsx:177]: fwiw it turns out track previews are implemented on the content end as 30s, so maybe easiest to update the UI side? Either way, maybe a comment/constant for the 15 here?

**Review by rickyrombo — APPROVED** [review]
Noticed on stage I was unable to change access from gated to public and have it stick? Is that related?

File: packages/web/src/pages/upload-page/fields/AccessAndSaleField.tsx (hunk 1 above)
Comment (amendelsohn) [reply packages/web/src/pages/upload-page/fields/AccessAndSaleField.tsx:177]: ah great call. I'll make it 30 before the end

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-access-sale-regression-fix

**Comment by amendelsohn** [comment]
> Noticed on stage I was unable to change access from gated to public and have it stick? Is that related?

Digging into this... looks like we're calling `confirmEditTrack` with `premium_conditions: null` and `is_premium: false`. Wondering if the `null` is getting dropped during the entity merge not replacing the premium conditions?

**Comment by rickyrombo** [comment]
> > Noticed on stage I was unable to change access from gated to public and have it stick? Is that related?
> 
> Digging into this... looks like we're calling `confirmEditTrack` with `premium_conditions: null` and `is_premium: false`. Wondering if the `null` is getting dropped during the entity merge not replacing the premium conditions?
> 
> When the track reloads later, it has `premium_conditions: { follower_id: 12345 }` but `is_premium: false`, and the dog ear is visible but not the "Special Access" label

That sounds like what's happening. https://github.com/AudiusProject/audius-protocol/blob/7b3120fd1da44563bd5530698ac91ab6c49a2594/discovery-provider/src/tasks/entity_manager/entities/track.py#L220-L226

Maybe we should send an empty object for premium conditions?

File: packages/web/src/pages/upload-page/fields/AccessAndSaleField.tsx [hunk 2]
```diff
@@ -109,12 +109,12 @@ const messages = {
...
     price: {
       tooLow: 'Price must be at least $0.99.',
-      tooHigh: 'Price must be less than $9.99.'
+      tooHigh: 'Price must be less than $9999.99.'
     },
     preview: {
       tooEarly: 'Preview must start during the track.',
       tooLate:
-        'Preview must start at lest 15 seconds before the end of the track.'
+        'Preview must start at lest 30 seconds before the end of the track.'
```
Comment (dylanjeffers) [inline packages/web/src/pages/upload-page/fields/AccessAndSaleField.tsx:117]: Least*

File: packages/web/src/pages/upload-page/fields/AccessAndSaleField.tsx [hunk 3]
```diff
@@ -109,12 +109,12 @@ const messages = {
   errors: {
     price: {
       tooLow: 'Price must be at least $0.99.',
-      tooHigh: 'Price must be less than $9.99.'
+      tooHigh: 'Price must be less than $9999.99.'
```
Comment (dylanjeffers) [inline packages/web/src/pages/upload-page/fields/AccessAndSaleField.tsx:112]: Lol

**Review by dylanjeffers — APPROVED** [review]
Awesome. Now just checking the usdc field is hidden in prod anyway right?

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-access-sale-regression-fix

**Comment by amendelsohn** [comment]
> Maybe we should send an empty object for premium conditions?

Oof sent `premium_conditions: {}` but still getting old `premium_conditions: { follower_id: 123 }` from discovery after confirmation.

**Comment by amendelsohn** [comment]
> Awesome. Now just checking the usdc field is hidden in prod anyway right?

Correct, though the other bug affects prod track edit

**Pull request merged**